core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala

```diff
@@ -169,4 +169,8 @@ object FileCommitProtocol extends Logging {
         ctor.newInstance(jobId, outputPath)
     }
   }
+
+  def getStagingDir(path: String, jobId: String): Path = {
+    new Path(path, ".spark-staging-" + jobId)
+  }
```

Comment:
@cloud-fan - wondering do you still think FileCommitProtocol V2 (https://issues.apache.org/jira/browse/SPARK-33298) is a pre-requisite for this PR? or this one is good to go separately, and we only need to worry about allowing adding prefix to file name per https://github.com/apache/spark/pull/30003 ?

Reply:
This PR is good to go as `getStagingDir` is defined in `object FileCommitProtocol`.

Reply:
HadoopMapReduceCommitProtocol is using the RDD.id as the job ID here, not uuid or hadoop job ID (which has had uniqueness issues elsewhere). if different jobs on different spark clusters try to write to same table using dynamic output partitioning, risk of output corruption -especially if hadoop "broken" v2 commit algorithm is used.